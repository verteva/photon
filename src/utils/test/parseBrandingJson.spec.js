import {
  parseBrandingJson,
  flattenObjectToCssVars,
} from '../parseBrandingJson';
import rawTheme from './rawTheme.json';

describe('parseBrandingJson.js', () => {
  it('parses a branding theme', () => {
    const result = parseBrandingJson(rawTheme);
    expect(result).toMatchSnapshot();
  });
});

describe('flattenObjectToCssVars.js', () => {
  it('parses a figma theme and then flattens it', () => {
    const result = parseBrandingJson(rawTheme);
    const cssResult = flattenObjectToCssVars(result);
    expect(cssResult).toMatchSnapshot();
  });
});
