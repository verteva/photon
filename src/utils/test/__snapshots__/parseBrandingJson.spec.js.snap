// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flattenObjectToCssVars.js parses a figma theme and then flattens it 1`] = `
Object {
  "--colors-brand-green": "#219653",
  "--colors-brand-orange": "orange",
  "--colors-brand-purple": "#6f53f9",
  "--colors-brand-red": "#eb5757",
  "--colors-brand-teal": "#31839b",
  "--colors-grey-10": "#f9f9f9",
  "--colors-theme-primary": "#f9f9f9",
  "--colors-theme-secondary": "#31839b",
  "--colors-theme-tertiary": "#eb5757",
  "--size-medium": 500,
  "--size-small": "500",
}
`;

exports[`parseBrandingJson.js parses a branding theme 1`] = `
Object {
  "colors": Object {
    "brand": Object {
      "green": "#219653",
      "orange": "orange",
      "purple": "#6f53f9",
      "red": "#eb5757",
      "teal": "#31839b",
    },
    "grey": Object {
      "10": "#f9f9f9",
    },
    "theme": Object {
      "primary": "#f9f9f9",
      "secondary": "#31839b",
      "tertiary": "#eb5757",
    },
  },
  "size": Object {
    "medium": 500,
    "small": "500",
  },
}
`;
